src/components/Project/DnDColumnList.tsx

```diff
@@ -0,0 +1,146 @@
+ */
+export default function DnDColumnList({ projectColumns }: DnDColumnListProps): JSX.Element {
+  const [columns, setColumns] = useState<ColumnDefinition[]>(projectColumns);
+  /** this state is passed via context to disable droppable on ticket area when column is dragged */
+  const [grabbedItem, setGrabbedItem] = useState<GrabbedItemCategory | null>(null);
+
+  // Set grabbed item to disable drop on unrelated place
+  const handleOnDragStart = ({ draggableId }: DragStart) => {
+    const grabbedItemCategory = draggableId.includes("column_")
+      ? GRABBED_ITEM_CATEGORY.COLUMN
+      : GRABBED_ITEM_CATEGORY.ROW_TICKET;
+
+    setGrabbedItem(grabbedItemCategory);
+  };
+
+  const handleOnDragEnd = ({ destination, source }: DropResult) => {
+    // Reset Grabbed item to make all items grabbable
+    if (!destination) {
+      setGrabbedItem(null);
+      return;
+    }
+
+    // When the target item is ticket
+    if (grabbedItem === GRABBED_ITEM_CATEGORY.ROW_TICKET) {
+      handleOnTicketDragEnd({ destination, source });
+      return;
+    }
+
+    const reorderedColumns = reorderSameListItems(columns, source.index, destination.index);
+    setNewColumns(reorderedColumns);
+  };
+
+  const handleOnTicketDragEnd = ({
+    destination,
+    source,
+  }: {
+    destination: DraggableLocation;
+    source: DraggableLocation;
+  }) => {
+    const sourceDroppableElementIdx = columns.findIndex((column) => column.id === Number(source.droppableId));
+    const destinationDroppableElementIdx = columns.findIndex(
+      (column) => column.id === Number(destination?.droppableId),
+    );
+    const isDroppedOnSameColumn = source.droppableId === destination?.droppableId;
+
+    if (isDroppedOnSameColumn) {
+      const targedColumn = columns[sourceDroppableElementIdx];
+      const reorderedTickets = reorderSameListItems(targedColumn.tickets, source.index, destination.index);
+
+      const newColumns = [...columns];
+      newColumns[sourceDroppableElementIdx].tickets = reorderedTickets;
+
+      setNewColumns(newColumns);
+      return;
+    }
+
+    // When dropped on another columns
+    const sourceColumn = columns[sourceDroppableElementIdx].tickets;
+    const destinationColumn = columns[destinationDroppableElementIdx].tickets;
+    const { resultSource, resultDestination } = moveItemToAnotherList(
+      sourceColumn,
+      destinationColumn,
+      source.index,
+      destination.index,
+    );
+    const newColumns = [...columns];
+    newColumns[sourceDroppableElementIdx].tickets = resultSource;
+    newColumns[destinationDroppableElementIdx].tickets = resultDestination;
+    setNewColumns(newColumns);
+  };
+
+  const setNewColumns = (newColumns: ColumnDefinition[]) => {
+    setGrabbedItem(null);
+    setColumns(newColumns);
+  };
+
+  return (
+    <GrabbedItemContext.Provider value={grabbedItem}>
+      <DragDropContext onDragEnd={handleOnDragEnd} onDragStart={handleOnDragStart}>
```

Comment:
This is the "central" DnD context across the app. For now, Any DnD action gets caught here and handled.